compiler-core/src/parse/token.rs

```diff
@@ -10,7 +10,7 @@ pub enum Token {
     Int { value: EcoString },
     Float { value: EcoString },
     String { value: EcoString },
-    CommentDoc { content: String },
+    CommentDoc { marker_start: u32, content: String },
```

Comment:
Why does this token have positional information? Positional information is handled at the level above, as ou can see with how none of the other tokens have any.

Reply:
I mostly just wanted a way to return the `marker_start` as well as the `start` without having to change a ton of signatures in the lexer (reasoning here https://github.com/gleam-lang/gleam/pull/3275#discussion_r1685511724 ). We could attempt the simplification mentioned in that reply instead if you'd prefer.